source/extensions/upstreams/tcp/generic/config.cc

```diff
@@ -16,7 +17,14 @@ TcpProxy::GenericConnPoolPtr GenericConnPoolFactory::createGenericConnPool(
     TcpProxy::TunnelingConfigHelperOptConstRef config, Upstream::LoadBalancerContext* context,
     Envoy::Tcp::ConnectionPool::UpstreamCallbacks& upstream_callbacks,
     StreamInfo::StreamInfo& downstream_info) const {
-  if (config.has_value()) {
```

Comment:
Could we have a test in tcp_proxy for this feature?

Reply:
tcp_proxy uses a generic upstream created by the generic connection pool factory, so this feature was added to the factory implementation. I added tests to the generic pool factory, LMK if you see any other required test case, IMO this should be sufficient.